core/cas-server-core-services-registry/src/main/java/org/apereo/cas/services/resource/AbstractResourceBasedServiceRegistry.java

```diff
@@ -387,9 +387,13 @@ protected RegisteredService getRegisteredServiceFromFile(final File file) {
     protected File getRegisteredServiceFileName(final RegisteredService service) {
         val fileName = resourceNamingStrategy.build(service, getExtensions()[0]);
         val svcFile = new File(this.serviceRegistryDirectory.toFile(), fileName);
+        try {
+            svcFile.getCanonicalPath();
+        } catch (final IOException e) {
+            throw new IllegalArgumentException("File name is invalid: " + e.getMessage(), e);
+        }
```

Comment:
This block of code is due to svcFile.getCanonicalPath() throwing an IOException when one of the tests passes in a filename with some characters that windows doesn't like. The lines are showing as not covered but that is only because the coverage is calculated on Ubuntu. I suppose to get coverage we would need a filename that throws IOException on Ubuntu. The getCanonicalPath() method is only called in this method in a debug statement so another alternative would be to change that to getName() or some other method that doesn't throw IOException. 

I think this is the only non-test code changed in this PR.

Reply:
This looks odd just by the looks of it, to use a method seemingly to just catch an exception, without using its return value. Could we use getPath or getAbsolutePath ?

Reply:
Yeah the test passes if the debug uses getAbsolutePath instead of getCanonicalPath so I changed it to that.